### PR TITLE
Update golangci-lint and config

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This also fixes the reported lints based on the new config.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/oci/runtime_vm.go [hunk 1]
```diff
@@ -273,7 +273,7 @@ func (r *runtimeVM) StartContainer(ctx context.Context, c *Container) error {
 		if err == nil {
 			// create a file on the exitsDir so that cri-o server can detect it
 			path := filepath.Join(r.exitsPath+"/", c.ID())
-			if fileErr := os.WriteFile(path, []byte("Exited"), 0o644); err != nil {
+			if fileErr := os.WriteFile(path, []byte("Exited"), 0o644); fileErr != nil {
```
Comment (saschagrunert) [inline internal/oci/runtime_vm.go:276]: Ouch, `err` is always `nil` here.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**